ocean_provider/validation/RBAC.py

```diff
@@ -0,0 +1,145 @@
+#
+# Copyright 2021 Ocean Protocol Foundation
+# SPDX-License-Identifier: Apache-2.0
+#
+
+import json
+
+from ocean_lib.common.agreements.service_types import ServiceTypesIndices
+from ocean_lib.web3_internal.transactions import sign_hash
+
+from ocean_provider.exceptions import RequestNotFound
+from ocean_provider.utils.basics import get_provider_wallet
+from ocean_provider.utils.util import msg_hash
+
+
+class RBACValidator:
+    def __init__(
+        self,
+        request_name,
+        request,
+    ):
+        self.request = request
+        action_mapping = self.get_action_mapping()
+        if request_name not in action_mapping.keys():
+            raise RequestNotFound("Request name is not valid!")
+        self.action = action_mapping[request_name]
+        self.credentials = {
+            "type": "address",
+            "address": get_provider_wallet().address,
+        }
+        self.component = "provider"
+        self.provider_address = get_provider_wallet().address
+
+    @staticmethod
+    def get_action_mapping():
+        return {
+            "EncryptRequest": "encryptUrl",
+            "InitializeRequest": "initialize",
+            "DownloadRequest": "access",
+            "ComputeRequest": "compute",
+            "ComputeStartRequest": "compute",
+        }
+
+    def fails(self):
+        return False
+
+    def get_request_dict(self):
```

Comment:
What is this, exactly?

Reply:
The request comes in this format: `{document: '<the actual request>'}`, so the value from the `document` key needs to be a dict instead of a string (I parse it with `json.dumps` in the tests), so basically that function converts from string to json.

Reply:
This doesn't seem to do what its name says. Is it also used only once? I am confused, let's have a call later and discuss.